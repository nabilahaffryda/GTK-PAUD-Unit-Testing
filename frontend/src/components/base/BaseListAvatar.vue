<template>
  <v-list>
    <v-list-item class="px-0">
      <v-list-item-avatar class="mr-1" :tile="tile" :color="color">
        <slot />
      </v-list-item-avatar>
      <v-list-item-content>
        <v-list-item-title class="link body-2" @click="onDetail" v-html="title"> </v-list-item-title>
        <v-list-item-subtitle class="caption" v-html="subtitle" />
      </v-list-item-content>
      <v-list-item-action v-if="useAction">
        <slot name="action" />
      </v-list-item-action>
    </v-list-item>
  </v-list>
</template>

<script>
export default {
  name: 'BaseListAvatar',
  props: {
    size: {
      type: Number,
      default: 32,
    },
    tile: {
      type: Boolean,
      default: false,
    },
    useAction: {
      type: Boolean,
      default: false,
    },
    color: {
      type: String,
      default: '',
    },
    caption: {
      type: String,
      default: '',
    },
    title: {
      type: String,
      required: true,
    },
    subtitle: {
      type: String,
      default: '',
    },
    detail: {
      type: Object,
      default: () => null,
    },
  },
  methods: {
    onDetail() {
      this.$emit('detail', this.detail);
    },
  },
};
</script>

<style scoped>
.link {
  white-space: normal;
  cursor: pointer;
}
</style>
