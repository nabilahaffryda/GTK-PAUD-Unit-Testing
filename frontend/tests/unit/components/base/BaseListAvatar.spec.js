import { mount, createLocalVue } from "@vue/test-utils";
import BaseListAvatar from "@/components/base/BaseListAvatar.vue";
import Vuetify from 'vuetify';
import Vue from 'vue';

Vue.use(Vuetify)
const localVue = createLocalVue();

describe('BaseListAvatar.vue', () => {
    let vuetify
    beforeEach(() => {
        vuetify = new Vuetify()
    })
    test('render list avatar', () => {
        const VAvatar = {
            props: [
                'size',
                'tile',
                'useAction',
                'color',
                'caption',
                'title',
                'subtitle',
                'detail'
            ],
            template: '<div><slot :tile="tile" /></div>'
        }

        const wrapper = mount(BaseListAvatar, {
            localVue,
            vuetify,
            stubs: {
                VAvatar
            },
            propsData: {
                title: 'avatar'
            }
        });
        expect(wrapper).toMatchSnapshot();
    })
})