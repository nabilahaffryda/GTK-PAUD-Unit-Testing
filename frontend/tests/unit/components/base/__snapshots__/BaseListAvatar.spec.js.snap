// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseListAvatar.vue render list avatar 1`] = `
<div role="list" class="v-list v-sheet theme--light">
  <div tabindex="-1" role="listitem" class="px-0 v-list-item theme--light">
    <div class="v-avatar v-list-item__avatar mr-1" style="height: 40px; min-width: 40px; width: 40px;"></div>
    <div class="v-list-item__content">
      <div class="v-list-item__title link body-2">avatar</div>
      <div class="v-list-item__subtitle caption"></div>
    </div>
    <!---->
  </div>
</div>
`;
